src/beamlime/handlers/timeseries_handler.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+import logging
+
+import scipp as sc
+
+from beamlime.handlers.to_nx_log import ToNXlog
+
+from ..core.handler import (
+    Config,
+    Handler,
+    HandlerFactory,
+    PeriodicAccumulatingHandler,
+)
+from ..core.message import MessageKey
+from .accumulators import ForwardingAccumulator, LogData
+
+
+class LogdataHandlerFactory(HandlerFactory[LogData, sc.DataArray]):
+    """
+    Factory for creating handlers for log data.
+
+    This factory creates a handler that accumulates log data and returns it as a
+    DataArray.
+    """
+
+    def __init__(
+        self,
+        *,
+        instrument: str,
+        logger: logging.Logger | None = None,
+        config: Config,
+        attribute_registry: dict[str, dict[str, any]],
```

Comment:
I'm a bit concerned that plain dictionaries are used in many places.
If it doesn't have to accept arbitrary fields, I think it'd better be a dataclass or pydantic models.

Reply:
At least the outer one should actually be a `Mapping`, I think. My thought was that attributes might be read from Kafka as well, similar to the config. Are you asking about the inner one?

Reply:
Ideally I'd like to forward the dicts to ScippNexus. Currently I don't think that is so straightforward, but given that the `attrs` in HDF5 are dict-like I am not convinced it should be changed here. Obviously using pydantic for validation or when we send this via Kafka makes sense.